src/slice.cr

```diff
@@ -566,6 +562,12 @@ struct Slice(T)
   protected def check_writable
     raise "Can't write to read-only Slice" if @read_only
   end
+
+  protected def check_size(count : Int)
```

Comment:
Why is this method `protected`?

Reply:
The only reason this is a`protected` method is to follow the same visibility of `check_writable` method. 
Do you suggest to make it private or public?

Reply:
I'm not sure why `check_writable` is `protected` either... I guess they could both be private.

Reply:
I'll update the visibility of those 2 methods to `private` then.

Reply:
Wait a minute, maybe someone else has input on this.

Reply:
There's `target.check_writable` in the code in `slice.cr`. That means it needs to be `protected`.

(I don't like the existing visibility rules, but that's what we have)

Reply:
If you don't need to do `obj.check_size` (with an explicit receiver) then there's no need to make it `protected`.

Reply:
Thank you @asterite for the explanation, I updated the visibility of `check_size` to be private.